core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
                 }
-                if ($this->modx->hasPermission('view_snippet')) {
-                    $this->searchSnippets();
+
+                
+                if($field != '*' AND in_array($field, $fields) !== true )
+                {
+                
+                    // Error: Unknown field
+                    $result['header'] = $modx->lexicon('uberbar_error_unknown_field_header');
+                    
+                    // You are searching inside an unknown field: <strong>{field}</strong>. Please remove it from your search and try something else.
+                    $s = array('{field}');
+                    $r = array($field);
+                    
+                    $result['msg'] = str_replace($s , $r , $modx->lexicon('uberbar_error_unknown_field_msg') );
+                    $result['options'] = '';
+                    
+                    return $result;
                 }
-                if ($this->modx->hasPermission('view_plugin')) {
-                    $this->searchPlugins();
+            }
+            return true;
+        }
+
+
+    /* : */
+        if($count == 2){
+            $validate = validate_Uberbar($elements,$query);
+            
+            if($validate === true){
+                // Return list of available resource types
+
+                
+                $result['header'] = $this->modx->lexicon('uberbar_pick_element_header');
+                
+                $f = array('{uberbar_mode}');
+                $r = array($uberbar_mode);                
+                
+                $result['msg'] = str_replace($f, $r , $this->modx->lexicon('uberbar_pick_element_msg'));
+                r:
+                $options = '';
+                foreach($elements as $key => $val){
+                    /* Only show elements you have permission to see. */
+                    if($val['permission'] === true) {
+                        $activeClass = ($key == $query[1]) ? 'active ' : '';
+                        $options .= '<li><a data-value="'.$key.'" class="'.$activeClass.$btn.'">'.$val['label'].'</a></li>';
+                    }
+                }           
+                $result['options'] = $options;
+                
+                $this->results[] = $result;
+            }
+            else{
+                $this->results[] = $validate;
+            }            
+        }
+        
+    /* :r: */
+        if($count == 3){
+            $validate = validate_Uberbar($elements,$query);
+            
+            if($validate === true){
+                // Return list of available fields that can be searched
+    
+                $result['header'] = $this->modx->lexicon('uberbar_pick_fieldname_header');
+                
+                $f = array('{uberbar_mode}');
+                $r = array($uberbar_mode);
+                
+                $result['msg'] = str_replace($f , $r , $this->modx->lexicon('uberbar_pick_fieldname_msg') );
```

Comment:
You can actually do this:

```` php
$result['msg'] = $this->modx->lexicon('uberbar_pick_fieldname_msg', array('uberbar_mode' => $uberbar_mode));
````
and in the lexicon refer to `[[+uberbar_mode]]` instead of `{uberbar_mode}`. I think it would probably be best to replace that to use the built-in approach rather than a manual str_replace.